backend/hasher/hasher.go

```diff
@@ -0,0 +1,550 @@
+//go:build !plan9 && !js
+// +build !plan9,!js
+
+// Package hasher implements a checksum handling overlay backend
+package hasher
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"path"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/rclone/rclone/fs"
+	"github.com/rclone/rclone/fs/cache"
+	"github.com/rclone/rclone/fs/config/configmap"
+	"github.com/rclone/rclone/fs/config/configstruct"
+	"github.com/rclone/rclone/fs/fspath"
+	"github.com/rclone/rclone/fs/hash"
+	"go.etcd.io/bbolt"
+)
+
+// Register with Fs
+func init() {
+	fs.Register(&fs.RegInfo{
+		Name:        "hasher",
+		Description: "Hasher will cache hashes",
+		NewFs:       NewFs,
+		CommandHelp: commandHelp,
+		Options: []fs.Option{{
+			Name:     "remote",
+			Required: true,
+			Help:     "Remote to cache hashes for, like myremote:mypath",
+		}, {
+			Name:     "hash_names",
+			Default:  "md5",
```

Comment:
You could use fs.CommaSepList or fs.SpaceSepList here if you wanted...

Reply:
For user convenience I wanted to split "comma optionally surrounded spaces or just by a space" which is easy on my eyes IMHO. Strictly speaking the regexp for such delimiter looks `\s*[,\s]\s*`. I simplified it to `[,\s]+` so splitting is a one-liner. This way a string like `a,b,,,c` or `a, b   ,c` will be split into slice `[a b c]`. I'm fine with such relaxed syntax.

If you wish, I can use your splitters resulting in more stringent syntax (where delimiter is exactly one comma or 1+ spaces but no comma). I don't care, just tell me

Reply:
or maybe I can add new `fs.CommaSpaceSepList` ?